server/middleware/__tests__/getEstablishmentFromUrl.spec.js

```diff
@@ -0,0 +1,27 @@
+const getEstablishmentFromUrl = require('../getEstablishmentFromUrl');
+
+describe('getEstablishmentFromUrl', () => {
+  const urlStub = 'content-hub.prisoner.service.justice.gov.uk/';
+  const next = jest.fn();
+  const res = {};
+  let req;
+  beforeEach(() => {
+    req = {
+      headers: {
+        host: `cookhamwood.${urlStub}`,
+      },
+      session: {},
+    };
+    next.mockClear();
+  });
+
+  it('should retrieve establishment from the header and set in the session', () => {
+    getEstablishmentFromUrl(req, res, next);
+    expect(req.session.establishmentName).toBe('cookhamwood');
+  });
+  it('should not retrieve establishment from the header and set in the session if not provided', () => {
+    req.headers.host = urlStub;
+    getEstablishmentFromUrl(req, res, next);
+    expect(req.session.establishmentName).not.toBeDefined();
+  });
```

Comment:
might be good to have tests for all the possible versions of the url?:

localhost:3000
wayland.prisoner-content-hub.local:3000
wayland-prisoner-content-hub-development-123.apps.live-1.cloud-platform.service.justice.gov.uk
wayland-prisoner-content-hub-staging.apps.live-1.cloud-platform.service.justice.gov.uk/
wayland-prisoner-content-hub-staging.apps.live-1.cloud-platform.service.justice.gov.uk/
content-hub.prisoner.service.justice.gov.uk
wayland.content-hub.prisoner.service.justice.gov.uk